### DISCUSSION TITLE
Feature request : Cash flow view Filter

### DISCUSSION THREAD

**Discussion by sdhawade** [ananthakumaran/paisa#21] [Ideas]
The cashflow view is a brilliant way to visualize the cash flows, especially the yearly view is very useful. Since I load all my transactions into paisa this even shows smaller cashflows of few thousands,  which makes the cashflow view quite noisy. A filter to show cashflows above a threshold will be very useful in my opinion to make this view even better.

**Comment by ananthakumaran** [comment]
I myself face this problem and I haven't found a good problem. Hierarchy was my first attempt to solve this, it improved things a bit, but not fully. The fundamental issue is that money can flow in both directions from liability/asset accounts and the Sankey diagram doesn't work very well if the flow is bidirectional. I have added a way to select the depth for income/expense, Let me know if it helps in your case.

![image](https://github.com/ananthakumaran/paisa/assets/149238/d36b17fd-adea-45f4-a71b-eb4429865173)

**Comment by dan335** [comment]
This seems to be missing in the latest release.  I like it.  What happened to it?  Any plans to put it back or was it a mistake?

**Reply by ananthakumaran** [reply]
@dan335 could you let me know what is missing and where you are looking at? I have removed the flat option from the master branch (not yet released, but you may find it missing in the demo). We will keep the hierarchy and the filter should be available as well.

**Reply by dan335** [reply]
Here's what it looks like for me.

![image](https://github.com/ananthakumaran/paisa/assets/499401/a276c4a0-e441-4bc0-8f4d-8e917e2f83a9)

It works in the demo but on my computer when I hover over it there is no popup for expenses and income.  One difference is that I have no income in my data only expenses.  Maybe that's why?

**Reply by dan335** [reply]
Glad to hear you are keeping it.  I'll try again when the next version is released.

**Reply by ananthakumaran** [reply]
If you have expense, it should at least show for expenses, I need to debug the issue then

**Comment by swarajsomala** [comment]
@ananthakumaran , I am also facing similar issue, it's looks so clumsy. Is it possible to add monthly monthly statement viewing features?

**Reply by ananthakumaran** [reply]
are you able to use the settings to control the granularity?
![image](https://github.com/ananthakumaran/paisa/assets/149238/3dd18c6a-bc07-4db9-a965-eadc731d3cc0)

**Reply by swarajsomala** [reply]
I can able to use it but it's not coming for each month, it showing for the entire year

**Reply by ananthakumaran** [reply]
Got it, month wise is not supported as of now, will think about it.
